superbench/benchmarks/micro_benchmarks/disk_performance.py

```diff
@@ -126,16 +126,52 @@ def add_parser_arguments(self):
                     help='Number of threads in %s test.' % io_str,
                 )
 
-    def _preprocess(self):
+        self._parser.add_argument(
+            '--verify',
+            type=str,
+            required=False,
+            help='Verification method specified for fio --verify flag.',
+        )
+
+    def _get_arguments_from_env(self):
+        """Read environment variables from runner used for parallel and fill in block_devices and numa.
+
+        Get 'PROC_RANK'(rank of current process) 'BLOCK_DEVICES' 'NUMA_NODES' environment variables
+        Get block_devices and numa according to 'BLOCK_DEVICES'['PROC_RANK'] and 'NUMA_NODES'['PROC_RANK']
+        - BLOCK_DEVICES should be comma-separated strings, each being colon-separated block devices.
+          E.g., "/dev/nvme0n1:/dev/nvme1n1,/dev/nvme2n1:/dev/nvme3n1".
+        - NUMA_NODES should be comma-separated numbers. E.g., "0,1".
+        Note: The config from env variables will overwrite the configs defined in the command line.
+        """
+        try:
+            if os.getenv('PROC_RANK'):
+                rank = int(os.getenv('PROC_RANK'))
+                if os.getenv('BLOCK_DEVICES'):
+                    self._args.block_devices = os.getenv('BLOCK_DEVICES').split(',')[rank].split(':')
+                if os.getenv('NUMA_NODES'):
+                    self._args.numa = int(os.getenv('NUMA_NODES').split(',')[rank])
+            return True
+        except BaseException as e:
+            self._result.set_return_code(ReturnCode.INVALID_ARGUMENT)
+            logger.error(
+                'The proc_rank is out of index of devices - benchmark: {}, message: {}.'.format(self._name, str(e))
+            )
+            return False
+
+    def _preprocess(self):    # noqa: C901
         """Preprocess/preparation operations before the benchmarking.
 
         Return:
             True if _preprocess() succeed.
         """
-        if not super()._preprocess():
+        if not super()._preprocess() or not self._get_arguments_from_env():
             return False
 
-        fio_path = os.path.join(self._args.bin_dir, self._bin_name)
+        fio_basic_command = os.path.join(self._args.bin_dir, self._bin_name)
+        if self._args.numa is not None:
+            fio_basic_command = f'numactl -N {self._args.numa} {fio_basic_command}'
+        if self._args.verify is not None:
+            fio_basic_command = f'{fio_basic_command} --verify={self._args.verify}'
```

Comment:
just curious if set verify, how to know the verify pass or not from the result?

Reply:
FIO will raise error message about which data blocks are corrupted, and it will be recorded in benchmark output. Also, the return code will be non-zero.